modular/executor/executor.go

```diff
@@ -83,16 +96,33 @@ func (e *ExecuteModular) eventLoop(ctx context.Context) {
 	}
 
 	statisticsTicker := time.NewTicker(time.Duration(e.statisticsOutputInterval) * time.Second)
+	updateSpTicker := time.NewTicker(3 * time.Second)
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		case <-statisticsTicker.C:
 			log.CtxInfo(ctx, e.Statistics())
+		case <-updateSpTicker.C:
+			sps, err := e.baseApp.Consensus().ListSPs(ctx)
+			if err != nil {
+				continue
+			}
+			e.mutex.RLock()
+			for _, sp := range sps {
+				e.spMap[sp.Id] = sp
+			}
+			e.mutex.RUnlock()
 		}
 	}
 }
 
+func (e *ExecuteModular) getSpByID(id uint32) *sptypes.StorageProvider {
+	e.mutex.RLock()
+	defer e.mutex.RUnlock()
+	return e.spMap[id]
```

Comment:
Maybe return nil and panic.
Refer to SPCachePool?

Reply:
There is no eviction so shouldn't be `nil`